core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -422,8 +422,8 @@ private ClusterCacheStatus initCacheStatusIfAbsent(String cacheName) {
          // (including the coordinator) before it starts on any node.
          AvailabilityStrategy availabilityStrategy;
          Configuration cacheConfiguration = cacheManager.getCacheConfiguration(cacheName);
-         boolean resolveConflictsOnMerge = cacheConfiguration.clustering().partitionHandling().mergePolicy() != null;
          PartitionHandling  partitionHandling = cacheConfiguration != null ? cacheConfiguration.clustering().partitionHandling().whenSplit() : null;
+         boolean resolveConflictsOnMerge = partitionHandling == PartitionHandling.ALLOW_READ_WRITES && cacheConfiguration.clustering().partitionHandling().mergePolicy() != null;
```

Comment:
I did think about that, however I thought it might be advantageous for us to be able to allow users to keep the existing merge behaviour, in the event a user depends on the old semantics, by simply specifying a null policy. That said I'm happy to remove this option if you think this is unnecessary.